keras_cv/applications/object_detection/retina_net/retina_net_test.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import tensorflow as tf
+
+import keras_cv
+
+
+class RetinaNetTest(tf.test.TestCase):
+    @pytest.fixture(autouse=True)
+    def cleanup_global_session(self):
+        # Code before yield runs before the test
+        yield
+        tf.keras.backend.clear_session()
+
+    def test_retina_net_construction(self):
+        retina_net = keras_cv.applications.RetinaNet(
+            classes=20,
+            bounding_box_format="xywh",
+            backbone="resnet50",
+            backbone_weights=None,
+            include_rescaling=True,
+        )
+        loss = keras_cv.losses.ObjectDetectionLoss(
+            classes=20,
+            classification_loss=keras_cv.losses.FocalLoss(
+                from_logits=True, reduction="none"
+            ),
+            box_loss=keras_cv.losses.SmoothL1Loss(l1_cutoff=1.0, reduction="none"),
+            reduction="auto",
+        )
+        retina_net.compile(
+            loss=loss,
+            optimizer="adam",
+            metrics=[
+                keras_cv.metrics.COCOMeanAveragePrecision(
+                    class_ids=range(20),
+                    bounding_box_format="xyxy",
+                    name="Standard MaP",
+                ),
+            ],
+        )
+
+        # TODO(lukewood) uncomment when using keras_cv.models.ResNet50
+        # self.assertIsNotNone(retina_net.backbone.get_layer(name="rescaling"))
```

Comment:
FWIW, I think this method raises an Error instead of returning None if the layer isn't found.